### PR TITLE
✨ add cluster.x-k8s.io/skip-defaulting-webhook

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**:

This PR adds a Cluster API standard annotation to enable optional defaulting webhooks.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: api/v1beta1/common_types.go [hunk 1]
```diff
@@ -142,6 +142,10 @@ const (
 	// will receive the resulting object.
 	TopologyDryRunAnnotation = "topology.cluster.x-k8s.io/dry-run"
 
+	// SkipDefaultingWebhook is an annotation that indicates we don't want to mutate objects during admission
+	// Ref: https://cluster-api.sigs.k8s.io/developer/providers/webhooks.html#defaulting-webhooks
```
Comment (fabriziopandini) [inline api/v1beta1/common_types.go:146]: @jackfrancis might be I'm missing something, but what is the use case for which we want to skip defaulting?

I'm asking because disabling a web-hook can lead to many unexpected edge-cases, and when we did something similar for the validation webhook we make sure this option works only when doing SSA dry-run calls (See [util func]( https://github.com/kubernetes-sigs/cluster-api/blob/bb287a1aabe1c1e54855441a10498086a9dd9683/util/topology/topology.go#L31)).

Comment (jackfrancis) [reply api/v1beta1/common_types.go:146]: Good question.

This is a draft PR that has some code to demonstrate how we might deal with this: https://github.com/kubernetes-sigs/cluster-api/issues/8668

tl;dr some gitops users don't want the yaml to be mutated upon admission, so that their yaml is able to be re-used as a source of truth

Comment (vincepri) [reply api/v1beta1/common_types.go:146]: 🤔 Sharing the same concerns Fabrizio pointed out above, it seems very error prone to disable any sort of webhook logic, most importantly ones that are effectively treated as defaults in the reconciliation code

**Comment by fabriziopandini** [comment]
@jackfrancis is it ok to close this PR?
if we want to move forward we probably need to discuss this in the issue first
